# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/credentialmanagement/identity/IdentityCredentialSerializationHelpers.h
# Commit: 63d9972bab1f
# Full Hash: 63d9972bab1f267a940ec30c7cea7a4c41e54fa6
# Author: Alexandru Marc <amarc@mozila.com>
# Date: 2025-06-06 08:52:03
# Regressor Bug: 1969017
# File Overlap Count: 2
# Description:
#   Revert "Bug 1969015, 1969017: apply code formatting via Lando" for causing leaks @ mochitest.toml
#   
#   This reverts commit a97cb1d6c08bc38e8644bfc916298b847dbac3a1.
#   
#   Revert "Bug 1969017 - Refactor FedCM to use a managed actor rather than tagging onto WindowGlobalParent - r=anti-tracking-reviewers,emz"
# ==============================================================================

diff -r 801e25cb4bfd -r 63d9972bab1f dom/credentialmanagement/identity/IdentityCredentialSerializationHelpers.h
--- a/dom/credentialmanagement/identity/IdentityCredentialSerializationHelpers.h	Thu Jun 05 18:51:36 2025 +0000
+++ b/dom/credentialmanagement/identity/IdentityCredentialSerializationHelpers.h	Thu Jun 05 21:40:49 2025 +0300
@@ -22,11 +22,21 @@
   static void Write(MessageWriter* aWriter, const paramType& aParam) {
     WriteParam(aWriter, aParam.mConfigURL);
     WriteParam(aWriter, aParam.mClientId);
+    WriteParam(aWriter, aParam.mOrigin);
+    WriteParam(aWriter, aParam.mLoginURL);
+    WriteParam(aWriter, aParam.mLoginTarget);
+    WriteParam(aWriter, aParam.mEffectiveQueryURL);
+    WriteParam(aWriter, aParam.mEffectiveType);
   }
 
   static bool Read(MessageReader* aReader, paramType* aResult) {
     return ReadParam(aReader, &aResult->mConfigURL) &&
-           ReadParam(aReader, &aResult->mClientId);
+           ReadParam(aReader, &aResult->mClientId) &&
+           ReadParam(aReader, &aResult->mOrigin) &&
+           ReadParam(aReader, &aResult->mLoginURL) &&
+           ReadParam(aReader, &aResult->mLoginTarget) &&
+           ReadParam(aReader, &aResult->mEffectiveQueryURL) &&
+           ReadParam(aReader, &aResult->mEffectiveType);
   }
 };
 
@@ -40,6 +50,11 @@
     WriteParam(aWriter, aParam.mNonce);
     WriteParam(aWriter, aParam.mLoginHint);
     WriteParam(aWriter, aParam.mDomainHint);
+    WriteParam(aWriter, aParam.mOrigin);
+    WriteParam(aWriter, aParam.mLoginURL);
+    WriteParam(aWriter, aParam.mLoginTarget);
+    WriteParam(aWriter, aParam.mEffectiveQueryURL);
+    WriteParam(aWriter, aParam.mEffectiveType);
   }
 
   static bool Read(MessageReader* aReader, paramType* aResult) {
@@ -47,7 +62,12 @@
            ReadParam(aReader, &aResult->mClientId) &&
            ReadParam(aReader, &aResult->mNonce) &&
            ReadParam(aReader, &aResult->mLoginHint) &&
-           ReadParam(aReader, &aResult->mDomainHint);
+           ReadParam(aReader, &aResult->mDomainHint) &&
+           ReadParam(aReader, &aResult->mOrigin) &&
+           ReadParam(aReader, &aResult->mLoginURL) &&
+           ReadParam(aReader, &aResult->mLoginTarget) &&
+           ReadParam(aReader, &aResult->mEffectiveQueryURL) &&
+           ReadParam(aReader, &aResult->mEffectiveType);
   }
 };
 
@@ -69,6 +89,11 @@
 };
 
 template <>
+struct ParamTraits<mozilla::dom::IdentityLoginTargetType>
+    : public mozilla::dom::WebIDLEnumSerializer<
+          mozilla::dom::IdentityLoginTargetType> {};
+
+template <>
 struct ParamTraits<mozilla::dom::CredentialMediationRequirement>
     : public mozilla::dom::WebIDLEnumSerializer<
           mozilla::dom::CredentialMediationRequirement> {};